test/tools/unified-spec-runner/entities.ts

```diff
@@ -266,7 +288,10 @@ export class UnifiedMongoClient extends MongoClient {
 
   // NOTE: pushCommandEvent must be an arrow function
   pushCommandEvent: (e: CommandEvent) => void = e => {
-    if (!this.isIgnored(e)) {
+    if (
+      (this.observeSensitiveCommands === true || !this.isSensitiveCommand(e)) &&
```

Comment:
If `observeSensitiveCommands` is off, and a sensitive command is located, why not push it into `IgnoredEvents`? There is precedent for this in the [unified spec runner](https://github.com/mongodb/node-mongodb-native/blob/9126b3ef38af1ba06bd7f7cbd7061144417a6d7b/test/tools/unified-spec-runner/entities.ts#L218C5-L220C6)

Reply:
To locate a sensitive command, we need to check if the `command` or `reply` properties of the command event an empty object (according to [spec](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst) and [other drivers implementation](https://github.com/mongodb/mongo-rust-driver/commit/114a519ac2fc2bc2bb8ff3cf765b029919d2c5cd#diff-fd8356adaae0c4a7d9d11585800aaf2ec0513c67795c2b66ec4bc3764b22885eR90-R91)):

"Note that the command and replies for such events will already have been redacted by the driver. If false or not specified, events for commands containing sensitive information MUST be ignored."

We don't have this info yet in the UnifiedMongoClient constructor. When `observeSensitiveCommands` is off, the hello can have or hasn't data in reply.

Reply:
Update: Talked about this comment thread over zoom; IgnoredEvents is about ignoring event types not individual events.